src/autorest-core/lib/configuration.ts

```diff
@@ -680,10 +741,23 @@ export class Configuration {
   public static async DetectConfigurationFile(fileSystem: IFileSystem, configFileOrFolderUri: string | null, messageEmitter?: MessageEmitter, walkUpFolders: boolean = false): Promise<string | null> {
     const originalConfigFileOrFolderUri = configFileOrFolderUri;
 
+    //
     if (!configFileOrFolderUri || configFileOrFolderUri.endsWith(".md")) {
       return configFileOrFolderUri;
     }
 
+    // try querying the Uri directly
+    if (ExistsUri(configFileOrFolderUri)) {
```

Comment:
required for example in case of gists... those Uri's don't necessarily end on `.md` but on some hex code